File: vue/src/api/position.js
import request from '@/utils/request'

export function PositionsList() {
  return request({
    url: '/position/list',
    method: 'post'
  })
}

export function getPositions() {
  return request({
    url: '/get/positions',
    method: 'post'
  })
}

export function addPosition(data) {
  return request({
    url: '/position/add',
    method: 'post',
    data: data
  })
}

export function delPosition(id) {
  return request({
    url: '/position/del',
    method: 'get',
    params: { id }
  })
}

export function updatePosition(data) {
  return request({
    url: '/position/update',
    method: 'post',
    data: data
  })
}

export function getHypos() {
  return request({
    url: '/get/hypos',
    method: 'post'
  })
}

File: go/bug/route/position.go
package route

import (
	"itflow/bug/handle"

	"github.com/hyahm/xmux"
)

var Position *xmux.GroupRoute

func init() {
	Position = xmux.NewGroupRoute("posistion")
	Position.Pattern("/position/list").Post(handle.PositionGet)
	Position.Pattern("/position/add").Post(handle.PositionAdd)
	Position.Pattern("/position/del").Get(handle.PositionDel)
	Position.Pattern("/position/update").Post(handle.PositionUpdate)
	Position.Pattern("/get/hypos").Post(handle.GetHypos)
	Position.Pattern("/get/positions").Post(handle.GetPositions)
}
